mlir-sycl/lib/Dialect/IR/SYCLOpsDialect.cpp

```diff
@@ -199,6 +200,91 @@ bool mlir::sycl::SYCLCastOp::areCastCompatible(::mlir::TypeRange Inputs,
   return HasArrayTrait && IsArray;
 }
 
+mlir::LogicalResult mlir::sycl::SYCLGEPOp::verify() {
+  // /* Available only when: (Dimensions > 0) */
+  // reference operator[](id<Dimensions> index) const;
+
+  // /* Available only when: (Dimensions > 1) */
+  // __unspecified__ operator[](size_t index) const;
+
+  // /* Available only when: (AccessMode != access_mode::atomic && Dimensions ==
+  // 1) */
+  // reference operator[](size_t index) const;
+  const auto AccessorTy = getOperand(0)
+                              .getType()
+                              .cast<mlir::MemRefType>()
+                              .getElementType()
+                              .cast<mlir::sycl::AccessorType>();
+
+  const auto Dimensions = AccessorTy.getDimension();
+  if (Dimensions == 0)
+    return emitOpError("dimensions cannot be zero");
+
+  const auto verifyResultType = [&]() -> mlir::LogicalResult {
+    const auto resultType = getResult().getType().dyn_cast<mlir::MemRefType>();
+
+    if (!resultType) {
+      return emitOpError("expecting memref return type. Got ") << resultType;
+    }
+
+    if (resultType.getElementType() != AccessorTy.getType()) {
+      return emitOpError(
+                 "expecting a reference to this accessor's value type. Got ")
+             << resultType;
```

Comment:
Suggest printing the expected type (AccessorTy.getType()) and the actual result type (resultType.getElementType()).